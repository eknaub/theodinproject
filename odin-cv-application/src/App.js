import { Component } from 'react';
import Header from './components/Header';
import Main from './components/Main';
import Footer from './components/Footer';
import './styles/app.css';

class App extends Component {
  render() {
    return (
      <div id='app'>
        <div id='header'><Header></Header></div>
        <div id='content'><Main></Main></div>
        <div id='footer'><Footer></Footer></div>
      </div>
    );
  };
}

export default App;
