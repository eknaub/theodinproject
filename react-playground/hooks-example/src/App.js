import React, { useEffect, useState } from "react";

function App() {
  const [count, setCount] = useState(0);
  const [color, setColor] = useState("black");

  const incrementCount = () => {
    setCount(count + 1);
  };

  useEffect(() => {
    // componentDidUpdate
    const changeColorOnClick = () => {
      if(color === "black") {
        setColor("red");
      }
      else {
        setColor("black");
      }
    };

    document.addEventListener("click", changeColorOnClick);

    return () => {
      // componentWillUnmount
      document.removeEventListener("click", changeColorOnClick);
    }
  }, [color]);

  useEffect(() => {
    // componentDidMount
  }, []);

  useEffect(() => {
    // componentDidMount and componentDidUpdate
  });

  return (
    <div>
      <div>{count}</div>
      <button onClick={incrementCount}>Increment</button>
      <div
        style={{
          color: "white",
          width: "100px",
          height: "100px",
          backgroundColor: color,
        }}
      >
        Change color.
      </div>
    </div>
  );
}

export default App;
