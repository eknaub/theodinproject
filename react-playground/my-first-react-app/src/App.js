import React, { Component } from 'react';
import MyComponent from './TestComponent';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0,
    };

    this.onClickBtn = this.onClickBtn.bind(this);
  }

  componentDidMount() {
    console.log("componentDidMount");
  }

  componentWillUnmount() {
    console.log("componentWillUnmount");
  }

  onClickBtn() {
    this.setState({
      count: this.state.count + 1,
    });
  }

  render() {
    return (
      <div>
        <MyComponent title={this.state.count} onButtonClicked={this.onClickBtn}></MyComponent>
      </div>
    );
  }
}

//Mit functions
/*
function App() {
  return (
    <div>
      <MyComponent title="Hello World" />
    </div>
  );
}
*/

export default App;
