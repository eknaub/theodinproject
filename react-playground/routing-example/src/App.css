nav {
  display: flex;
  justify-content: space-around;
  align-items: center;
  min-height: 10vh;
  background: rgb(67, 73, 75);
  color: white;
}

.nav-links {
  width: 40%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  list-style: none;
}