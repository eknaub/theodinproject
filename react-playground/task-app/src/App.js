import { Component } from 'react';
import Overview from './components/Overview';
import uniqid from "uniqid";

class App extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      task: { 
        text: '',
        id: uniqid(),
        count: 1,
      },
      tasks: [],
    }

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
  };

  handleChange(e) {
    this.setState({
      task: {
        text: e.target.value,
        id: this.state.task.id,
        count: this.state.task.count,
      }
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState( {
      tasks: [...this.state.tasks, {
        task: this.state.task,
        id: this.state.task.id,
      }],
      task: { 
        text: '',
        id: uniqid(),
        count: this.state.task.count + 1,
      },
    });
  }

  handleDelete(id) {
    const newList = this.state.tasks.filter((item) => item.id !== id);
    this.setState( {
      tasks: newList,
    });
  }

  render() {
    const { task, tasks } = this.state;

    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <label htmlFor="taskInput">Todo</label>
          <input type='text' id='taskInput' value={task.text} onChange={this.handleChange}></input>
          <button type='submit'>Add</button>
        </form>
        <Overview tasks={tasks} handleDelete={this.handleDelete}></Overview>
      </div>
    )
  }
}

export default App;
