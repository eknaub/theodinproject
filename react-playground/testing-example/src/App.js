import React, { useState } from "react";

const App = () => {
  const [heading, setHeading] = useState("Initial");

  const clickHandler = () => {
    setHeading("clicked");
  }

  return (
    <>
      <button type="button" onClick={clickHandler}>
        Click me
      </button>
      <h1>{heading}</h1>
    </>
  )
}

export default App;
