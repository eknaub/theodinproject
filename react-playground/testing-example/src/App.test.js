import { render, screen } from '@testing-library/react';
import App from './App';
import React from "react";
import userEvent from "@testing-library/user-event";

describe("App component", () => {
  it("renders heading inital", () => {
    const { container } = render(<App/>);
    expect(container).toMatchSnapshot();
  })

  it("renders heading after click", () => {
    render(<App/>);
    const button = screen.getByRole("button", {name: "Click me"});

    userEvent.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/clicked/);
  })
})
