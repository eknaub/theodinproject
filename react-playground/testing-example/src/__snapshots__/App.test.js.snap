// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders heading inital 1`] = `
<div>
  <button
    type="button"
  >
    Click me
  </button>
  <h1>
    Initial
  </h1>
</div>
`;
